File: src/main/java/nl/tom/hstone/service/CardComparator.java
package nl.tom.hstone.service;

import java.util.Comparator;

import nl.tom.hstone.model.Card;

/**
 * Compares cards by cardId.
 * 
 * @author Tom
 *
 */
public class CardComparator implements Comparator<Card> {

	SortDirection dir;

	public CardComparator(SortDirection dir) {
		this.dir = dir;
	}

	@Override
	public int compare(Card card1, Card card2) {
		// TODO null safe
		return dir.getMultiplier() * card1.getCardId().compareTo(card2.getCardId());
	}
}

File: app/src/main/java/com/krayem/hearthstone/ui/main/CardComparator.kt
package com.krayem.hearthstone.ui.main

import com.krayem.hearthstone.model.Card

class CardComparator : Comparator<Card> {

    override fun compare(a: Card?, b: Card?): Int {
        return when {
            a?.name != null && b?.name != null -> a.name.compareTo(b.name)
            else -> 0
        }
    }
}